staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go

```diff
@@ -278,6 +284,7 @@ func (f *ConfigFlags) toDiscoveryClient() (discovery.CachedDiscoveryInterface, e
 	httpCacheDir := filepath.Join(cacheDir, "http")
 	discoveryCacheDir := computeDiscoverCacheDir(filepath.Join(cacheDir, "discovery"), config.Host)
 
+	// TODO(negz): Should the discovery cache time be configurable?
 	return diskcached.NewCachedDiscoveryClientForConfig(config, discoveryCacheDir, httpCacheDir, time.Duration(10*time.Minute))
```

Comment:
I wonder if this should be a fixed duration; 15 seconds of delay with thousands of CRDs is pretty good but it's still not ideal to hit delays every 10 mins if we don't need to.

Reply:
> 15 seconds of delay with thousands of CRDs is pretty good

do you know where that time is being spent?

Reply:
That's because discovery still happens for the 1900 CRDs. It's not disabled, just the client doesn't do any rate limiting anymore, only the server.

Reply:
> I wonder if this should be a fixed duration; 15 seconds of delay with thousands of CRDs is pretty good but it's still not ideal to hit delays every 10 mins if we don't need to.

It is something I've been also considering.